src/Extension/Core/DataTransformer/MoneyToLocalizedStringTransformer.php

```diff
@@ -131,68 +129,95 @@ public function reverseTransform($value)
+    {
+        $currency = $this->defaultCurrency;
+        $locale = \Locale::getDefault();
+
+        if (!isset(self::$patterns[$locale])) {
+            self::$patterns[$locale] = [];
+        }
+
+        if (!isset(self::$patterns[$locale][$currency])) {
+            $formatter = new \NumberFormatter($locale, \NumberFormatter::CURRENCY);
+            $pattern = $formatter->formatCurrency(123.00, $currency);
+
+            // 1=left-position currency, 2=left-space, 3=right-space, 4=left-position currency.
+            // With non latin number scripts.
+            preg_match(
+                '/^([^\s\xc2\xa0]*)([\s\xc2\xa0]*)\p{N}{3}(?:[,.]\p{N}+)?([\s\xc2\xa0]*)([^\s\xc2\xa0]*)$/iu',
+                $pattern,
+                $matches
+            );
+
+            if (!empty($matches[1])) {
+                self::$patterns[$locale][$currency] = ['%1$s'.$matches[2].'%2$s', $matches[1]];
+            } elseif (!empty($matches[4])) {
+                self::$patterns[$locale][$currency] = ['%2$s'.$matches[3].'%1$s', $matches[4]];
+            } else {
+                throw new \InvalidArgumentException(
+                    sprintf('Locale "%s" with currency "%s" does not provide a currency position.', $locale, $currency)
                 );
             }
         }
 
-        // NumberFormatter::parse() does not round
-        $result = $this->round($result);
-        $result *= $this->divisor;
-
-        if (false === $currency) {
-            $currency = $this->defaultCurrency;
-        }
-
-        return new MoneyValue($currency, (string) $result);
+        return sprintf(self::$patterns[$locale][$currency][0], self::$patterns[$locale][$currency][1], $value);
     }
 
-    private function getNumberFormatter(int $type = \NumberFormatter::CURRENCY): \NumberFormatter
+    /**
+     * Removes the currency symbol.
+     *
+     * ICU cannot format() with currency, as this
+     * produces a number with the `¤` symbol.
+     * And, decimal doesn't include scale when 0.
+     *
+     * @param string $value
+     *
+     * @return string
+     */
+    private function removeCurrencySymbol(string $value): string
     {
-        $formatter = new \NumberFormatter(\Locale::getDefault(), $type);
+        $currency = $this->defaultCurrency;
+        $locale = \Locale::getDefault();
 
-        if (null !== $this->scale) {
-            $formatter->setAttribute(\NumberFormatter::FRACTION_DIGITS, $this->scale);
-            $formatter->setAttribute(\NumberFormatter::ROUNDING_MODE, $this->roundingMode);
+        if (!isset(self::$patterns[$locale][$currency])) {
+            // Initialize the cache, ignore return.
+            $this->addCurrencySymbol('123');
         }
 
-        $formatter->setAttribute(\NumberFormatter::GROUPING_USED, $this->grouping);
-
-        return $formatter;
+        return preg_replace('#(\s?'.preg_quote(self::$patterns[$locale][$currency][1], '#').'\s?)#u', '', $value);
```

Comment:
This can fail when the value's currency is different from the default one (exporting can be done later).